src/core/net/ip6.cpp

```diff
@@ -1103,6 +1109,10 @@ Error Ip6::ProcessReceiveCallback(Message &          aMessage,
 
     mReceiveIp6DatagramCallback(message, mReceiveIp6DatagramCallbackContext);
 
+#if OPENTHREAD_CONFIG_IP6_BR_COUNTERS_ENABLE
+    UpdateBorderRoutingCounters(header, messageLength, false);
```

Comment:
Can consider using `aMessage.GetLength()` directly here (so we don't need to define `messageLength` variable)?

Reply:
I used this variable because I was not sure if message object would get modified or destructed in the function body. Can we assume that the message is still available at this point?